spec/runtime/setup_spec.rb

```diff
@@ -857,6 +857,42 @@ def clean_load_path(lp)
     expect(out).to eq("true\ntrue")
   end
 
+  context "with bundler is located in symlinked GEM_HOME", :focus => true do
+    let(:gem_home) { Dir.mktmpdir }
+    let(:symlinked_gem_home) { Tempfile.new }
+    let(:bundler_dir) { File.expand_path("../..", __dir__) }
+
+    before do
+      FileUtils.ln_sf(gem_home, symlinked_gem_home)
+      gems_dir = File.join(gem_home, "gems")
+      specifications_dir = File.join(gem_home, "specifications")
+      Dir.mkdir(gems_dir)
+      Dir.mkdir(specifications_dir)
+
+      FileUtils.ln_s(bundler_dir, File.join(gems_dir, "bundler-#{Bundler::VERSION}"))
+
+      gemspec = File.read("#{bundler_dir}/bundler.gemspec").
+                sub("Bundler::VERSION", %("#{Bundler::VERSION}"))
+      gemspec = gemspec.lines.grep_v(%r{lib/bundler/version}).join
+
+      File.open(File.join(specifications_dir, "bundler.gemspec"), "wb") do |f|
+        f.write(gemspec)
+      end
+    end
+
+    it "should succesfully require 'bundler/setup'" do
+      install_gemfile ""
+
+      ENV["GEM_PATH"] = symlinked_gem_home.to_path
+      ruby <<-R
+        $LOAD_PATH.delete_at(0)
+        puts (require 'bundler/setup')
```

Comment:
imho we should add something like `raise "missing bundler lib path in #{$LOAD_PATH}" unless $LOAD_PATH.find ...`